include/revng/Pypeline/ObjectIDImpl.h

```diff
@@ -0,0 +1,138 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <set>
+
+#include "boost/functional/hash.hpp"
+
+#include "llvm/ADT/StringRef.h"
+
+#include "revng/Model/TypeDefinition.h"
+#include "revng/Support/MetaAddress.h"
+
+class ObjectID {
+private:
+  using TypeKey = model::TypeDefinition::Key;
+  std::variant<std::monostate, MetaAddress, TypeKey> Component;
+  friend constexpr std::strong_ordering operator<=>(const ObjectID &,
+                                                    const ObjectID &);
+
+public:
+  enum Kind {
+    Root = 0,
+    Function = 1,
+    Type = 2
```

Comment:
`TypeDefinition`. Check the whole patchset.

Reply:
Fixed